src/Joonasw.AspNetCore.SecurityHeaders/ReferrerPolicy/ReferrerPolicyMiddleware.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Options;
+
+namespace Joonasw.AspNetCore.SecurityHeaders.ReferrerPolicy
+{
+    public class ReferrerPolicyMiddleware
+    {
+        private const string HeaderName = "Referrer-Policy";
+        private readonly RequestDelegate _next;
+        private readonly string _headerValue;
+
+        public ReferrerPolicyMiddleware(RequestDelegate next, IOptions<ReferrerPolicyOptions> options)
+        {
+            _next = next;
+            _headerValue = options.Value.ReferrerPolicyValue.DefaultValue();
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            context.Response.Headers.Add(HeaderName, _headerValue);
```

Comment:
Need to add a check to not add the header if it has already been added to the response.